jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js

```diff
@@ -193,56 +187,65 @@ YUI.add('user-profile-model-list', function() {
+      }, () => {
+        const callback = this._destroyModelCallback.bind(
+          this, uuid, model.name);
+        this.props.destroyModels([uuid], callback);
+      });
     },
 
     /**
       Callback for the controller delete model call.
 
       @method _destroyModelCallback
+      @param {String} uuid the model being deleted. Bound when the callback is
+        passed into the API call.
+      @param {String} modelName the name of the model being deleted; primarily
+        useful for providing user-friendly notifications messages. Bound when
+        the callback is passed into the API call.
       @param {String} err The error from the request, or null.
-      @param {Array} modelList The list of models.
+      @param {Object} results The result for the model being deleted. The
+        object is keyed to the model UUID.
     */
-    _destroyModelCallback: function(err, results) {
+    _destroyModelCallback: function(uuid, modelName, err, results) {
       const addNotification = this.props.addNotification;
-      // Handle global errors.
-      if (err) {
+      // Handle global errors or model-specific errors.
+      const error = err || results[uuid];
+      if (error) {
         addNotification({
           title: 'Model destruction failed',
-          message: 'The model failed to be destroyed: ' + err,
+          message: `Could not destroy model "${modelName}": ${error}`,
           level: 'error'
         });
-        return;
-      }
-      // Ignore all but the first UUID because the UI only allows deleting one
-      // at a time.
-      const uuid = Object.keys(results)[0];
-      // Check for model-specific errors.
-      const uuidError = results[uuid];
-      if (uuidError) {
+      } else {
+        // Handle a successful deletion.
         addNotification({
-          title: 'Model destruction failed',
-          message: 'The model failed to be destroyed: ' + uuidError,
-          level: 'error'
+          title: 'Model destroyed',
+          message: `The model "${modelName}" is destroyed.`,
+          level: 'important'
         });
-        return;
+        // XXX kadams54: Ideally the model would change in the DB and that would
+        // trigger a re-render. Right now we're getting the data from an API;
+        // eventually, once we re-write the model layer to move away from YUI,
+        // this will hopefully change to become more React-friendly. Until then
+        // this hack will do.
+        const destroyedModel = this.state.modelList.find(model => {
+          return model.uuid === uuid;
+        });
+        destroyedModel.isAlive = false;
       }
-      // Handle a successful deletion.
-      addNotification({
-        title: 'Model destroyed',
-        message: 'The model is currently being destroyed.',
-        level: 'important'
+      // Remove the model UUID from the list of in-flight deletion requests.
+      const destroyingModels = this.state.destroyingModels.filter(id => {
```

Comment:
do we need to do this as well in case of an error? It's not still destroying any more if we've got an error condition is it?